### ISSUE TITLE
Meaning of features_dc and features_rest

### ISSUE THREAD

**Issue by robofar** [graphdeco-inria/gaussian-splatting#728]
Hello. I know that color of 3D Gaussian in this paper is represented by spherical harmonics (SH). By looking at `gaussian_splatting.py` I see there are two attributes of class `GaussianModel`:

- `features_dc`
- `features_rest`

I suppose these attributes represent color and spherical harmonics somehow, but can someone tell me why we have two attributes, and what each of them represent? I couldn't infer 100% from the code itself

Also code from `gaussian_model.py` specifically method `create_from_pcd` is having this part of code:
```
fused_color = RGB2SH(torch.tensor(np.asarray(pcd.colors)).float().cuda())
features = torch.zeros((fused_color.shape[0], 3, (self.max_sh_degree + 1) ** 2)).float().cuda()
features[:, :3, 0 ] = fused_color
features[:, 3:, 1:] = 0.0
```

followed by:
```
self._features_dc = nn.Parameter(features[:,:,0:1].transpose(1, 2).contiguous().requires_grad_(True))
self._features_rest = nn.Parameter(features[:,:,1:].transpose(1, 2).contiguous().requires_grad_(True))
```

Can someone explain:

1. What is fused_color
2. What features represent (and why one part is assigned fused_color another 0)
3. What `features_dc` and `features_rest` are from this context

**Comment by altaykacan** [comment]
Hey!

I'm no expert on this but as far as I can tell when dealing with direction dependent lighting effects spherical harmonics are used. These are special functions that can be combined to represent any function on the surface of a sphere, which in our case would be the direction dependent appearance of a given Gaussian. In my understanding, the higher the degree the higher frequencies (finer details) can be represented.

So I'm guessing `features_dc` represents _diffuse color_ which is the kind of the base color of the object if you're not considering any lighting effects. This would correspond to the lowest degree of the spherical harmonics where it is just constant (no view dependence). With this interpretation `features_rest` would, well, be the rest of the spherical harmonic coefficients.

I also came upon this helpful source during my study, maybe it helps you too: https://3dvar.com/Green2003Spherical.pdf

I hope this helps! :)

**Comment by robofar** [comment]
From my basic understanding, your comment makes sense, and names of variables correspond to your explanation. Thank you!

**Comment by yuedajiong** [comment]
in this algorihtm:  
1. degree from 0(only dc no rest) to 3, you can see: from lower to higher frequencies/detials
2. different learning rate, lower lr for rest
